### PR TITLE
checkout kubernetes cluster supported feature

### PR DESCRIPTION
**Pull request by zouyee**
https://github.com/kubernetes/kube-state-metrics/issues/212

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kube-state-metrics/220)

### CODE REVIEW

**Comment by zouyee** [comment]
@brancz

**Comment by brancz** [comment]
Nice! For backward compatibility we will have to put this behind a flag and only use it when explicitly enabled. Furthermore I'd prefer to introduce a map that maps the api group + version to the collector identifier, that way we can keep using the available collectors list for the `--collectors=...` flag as used previously, and only have to change a single string in case an API is promoted/deprecated.

**Comment by WIZARD-CXY** [comment]
hmm agreed with @brancz

File: main.go [hunk 1]
```diff
@@ -43,34 +45,34 @@ const (
...
-		"nodes":                  collectors.RegisterNodeCollector,
-		"pods":                   collectors.RegisterPodCollector,
-		"replicasets":            collectors.RegisterReplicaSetCollector,
-		"replicationcontrollers": collectors.RegisterReplicationControllerCollector,
-		"resourcequotas":         collectors.RegisterResourceQuotaCollector,
-		"services":               collectors.RegisterServiceCollector,
-		"statefulsets":           collectors.RegisterStatefulSetCollector,
-		"persistentvolumeclaims": collectors.RegisterPersistentVolumeClaimCollector,
+		"cronjobs/v2alpha1":         collectors.RegisterCronJobCollector,
+		"daemonsets/v1beta1":        collectors.RegisterDaemonSetCollector,
```
Comment (andyxning) [inline main.go]: Hard code the version for each group here is fragile and maybe diverse from the actually used version in each register function. Maybe we should find another way to correlate the group version used in register function and the keys used in `availableCollectors`.

Comment (zouyee) [inline main.go]: now I add `init function` in each collector.

Comment (brancz) [inline main.go]: Could you please find another solution for this @zouyee ? So far we've been able to keep global things to a minimum, but 14 additional init functions just makes this overly complex to maintain.

Comment (zouyee) [inline main.go]: I have restored to the original state, used `variable` to shown the GroupVersionResource of collector.

**Comment by zouyee** [comment]
@brancz  PTAL

**Comment by andyxning** [comment]
@zouyee Could you please resolve the conflict. We need this feature and after conflict is resolved we can continue to review.

**Comment by zouyee** [comment]
I will fix it!

**Pull request closed without merging**